backend/api/routes.py

```diff
@@ -96,11 +96,34 @@ def get_values():
     start_time = request.args.get('start_time')
     end_time = request.args.get('end_time')
 
-    return Values.get(tuple(point_ids), start_time, end_time)
+    search = request.args.get('search') or 'TRUE'
+
+    return Values.get(tuple(point_ids), start_time, end_time, search) or "[]"
 
 
 @api.route('/values', methods=['POST'])
 def post_values():
     Values.add(request.get_json())
 
     return "Success"
+
+
+@api.route('/points/verify')
+def search_verify():
+    if request.args.get("search") is None:
+        abort(400, "Request must include a `search` argument")
+    try:
+        return Points.counts_where(Search.points(request.args.get("search"))) or "[]"
+    except InvalidSearchException as e:
+        return "Invalid Point Search: " + str(e)  # TODO: Is this the right way to return the error?
```

Comment:
Hmm... I'm not sure either, but my thought is that if an invalid search is created it's the UI's fault, not the user's, so we should throw a 400 (actually my thought would be do this regardless of fault). Then the UI can choose how and if to surface it to the user. But any way is fine as long as we handle it in the UI

Reply:
Hm, I hadn't thought of it that way. I was thinking that 400 means bad request, and there's nothing wrong with the HTTP request, just something wrong with the content. And I was thinking of it as user error, "user" being whatever person or software was using the API. But you have a good point, that probably the only thing which would cause an `InvalidSearchException` would be a problem with the UI. In fact, if there was a user error, like mismatched parenthesis, that would be thrown as a Psycopg2 error, that would not be caught, and the response would be a 500. So I am no more certain about what to do in the event of an `InvalidSearchException`, but we definitely should be catching any Psycopg2 errors.